cognite/seismic/protos/v1/seismic_service_datatypes.proto

```diff
@@ -281,3 +281,51 @@ enum SurveyCoverageSource {
     // Requesting or returning calculated coverage
     CALCULATED = 2;
 }
+
+message GeometryFilter {
+    Geometry geometry = 1;
+    InterpolationMethod interpolation_method = 2; // Only used for linear geometries
+}
+
+message TraceHeaderField {}
+
+message Seismic2dExtent {
+    TraceHeaderField trace_key = 1; // Must be an applicable field for 2D
+    LineDescriptor trace_range = 2; // Range of header values to use
+}
+
+message Seismic3dExtent {
+    oneof type {
+        Seismic3dRects rects = 1;
+        Seismic3dDef def = 2;
+    }
+}
+
+message Seismic3dRects {
+    repeated Seismic3dRect rects = 1;
+}
+
+message Seismic3dRect {
+    LineDescriptor inline_range = 1;
+    LineDescriptor xline_range = 2;
+}
+
+message Seismic3dDef {
+    TraceHeaderField major_direction = 1; // Must be inline or crossline
+    map<int32, LineDefs> lines = 3;
+}
+
+message LineDefs {
```

Comment:
Could we rename LineDef to, say, LineFragment, LineSegment, or something?

Reply:
Or if we just use repeated LineDescriptor... then that's an OK name I guess.

Reply:
And then there's the question of geometry based queries. Not that we need to support that on the first go, but we should have a plan for where that gets bolted on. 
Would that be inside Seismic3dDef or an alternative message? Seismic3DGeometry?

Reply:
I believe this will be fine for covering the use case supported by the old trace streaming API though - which is great! =)